amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/audit/ElectronicRecordAuditLogTest.java

```diff
@@ -0,0 +1,244 @@
+    }
+
+    @Test(description = "Audit log for newly filed electronic record")
+    @AlfrescoTest(jira="RM-4303")
+    public void newElectronicRecordAudit() {
+        List<AuditEntry> auditEntries= auditLog.getRMAuditLogAll(getAdminUser().getUsername(),getAdminUser().getPassword(),100);
+
+        // newly created record contains 2 events: "file to" and metadata update
+        // the order in which object creation and metadata update are listed isn't always identical due to
+        // both happening in the same transaction
+        assertTrue("File To Event is not present.",auditEntries.stream().anyMatch(x -> x.getEvent().startsWith("File to")));
+        assertTrue("Updated metadata Event is not present.",auditEntries.stream().anyMatch(x -> x.getEvent().startsWith("Updated Metadata")));
+    }
+
+    @Test
+        (
+            dependsOnMethods = "newElectronicRecordAudit",
+            description = "Viewing electronic record audit log is itself an auditable event"
+        )
+    @AlfrescoTest(jira="RM-4303")
+    public void electronicRecordAuditIsEvent()
+    {
+        List<AuditEntry> auditEntries= auditLog.getRMAuditLogAll(getAdminUser().getUsername(),getAdminUser().getPassword(),100);
+        assertTrue("Audit View Event is not present.",auditEntries.stream().anyMatch(x -> x.getEvent().startsWith("Audit View")));
+    }
+
+    @Test
+        (
+            dependsOnMethods = "electronicRecordAuditIsEvent",
+            description = "Rename electronic record is an edit metadata event"
+        )
+    @AlfrescoTest(jira="RM-4303")
+    public void renameElectronicRecord() {
+        auditLog.clearAuditLog(rmAdmin.get().getUsername(),rmAdmin.get().getPassword());
+        Record renameElectronicRecord = createRecordModel("edited " + electronicRecord.getName(), "", "");
+
+        // rename record
+        getRestAPIFactory().getRecordsAPI().updateRecord(renameElectronicRecord, electronicRecord.getId());
+        assertStatusCode(OK);
+
+        // we expect 1 new event: "metadata update"
+        List<AuditEntry> auditEntries= auditLog.getRMAuditLogAll(getAdminUser().getUsername(),getAdminUser().getPassword(),100);
+        assertTrue("Updated metadata Event is not present.",auditEntries.stream().anyMatch(x -> x.getEvent().startsWith("Updated Metadata")));
+    }
+
+    @Test (
+        dependsOnMethods = "newElectronicRecordAudit",
+        description = "Complete and reopen electronic record")
+    @AlfrescoTest(jira="RM-4303")
+    public void completeAndReopenElectronicRecord() {
+        electronicRecord2 = createElectronicRecord(recordFolder1.getId(),AUDIT_COMPLETE_REOPEN_ELECTRONIC_RECORD);
+
+        // complete record
+        recordApi.completeRecord(rmAdmin.get().getUsername(),rmAdmin.get().getPassword(),
+            electronicRecord2.getName());
+
+        try
+        {
+            Utility.sleep(1000, 30000, () ->
```

Comment:
I didn't see what we're waiting for here (and in the test below)

Reply:
The RecordsAPI's completeRecord method is making the POST request to RM_ACTIONS_API "{0}rma/actions/ExecutionQueue" which takes the request and then it can take sometime before the aspect is updated to "rma:declaredRecord" hence we are waiting for the Aspect to be updated.